### PR TITLE
Supports day format in ReadableDuration (#3687)

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
Cherry pick #3687 and adjust a little to make it backward compatible

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
manual test

## Does this PR affect documentation (docs) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

File: src/util/config.rs [hunk 1]
```diff
@@ -452,6 +454,11 @@ impl Serialize for ReadableDuration {
     {
         let mut dur = util::time::duration_to_ms(self.0);
         let mut buffer = String::new();
+        // For backward compatibility, do not render config that old version cannot parse.
+        // if dur >= DAY {
```
Comment (BusyJay) [inline src/util/config.rs]: Remove the commented out implementation.

File: src/util/config.rs [hunk 2]
```diff
@@ -1089,6 +1097,7 @@ mod test {
             (0, 0, "0s"),
             (0, 1, "1ms"),
             (2, 0, "2s"),
+            (24 * 3600, 0, "24h"),
```
Comment (BusyJay) [inline src/util/config.rs]: Should add day test case.

File: src/util/config.rs (hunk 1 above)
Comment (Connor1996) [reply src/util/config.rs]: It is a cherry pick. If we just delete them, then it is hard to know what we change and why it is not consistent with the master branch.

File: src/util/config.rs (hunk 2 above)
Comment (zhangjinpeng1987) [reply src/util/config.rs]: @Connor1996 How about add a test case like (3600 * 24, 0, "1d")?

**Pull request merged**